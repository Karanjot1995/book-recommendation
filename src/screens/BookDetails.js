import {HOST} from '@env';
import { useNavigation } from "@react-navigation/native";
import React, { useState, useEffect } from "react";
import {Modal,Image, Text, StyleSheet, View , Pressable , TouchableOpacity , TextInput,FlatList, ScrollView} from "react-native";
import { Dimensions } from "react-native";
const win = Dimensions.get('window');
import { MaterialCommunityIcons } from "@expo/vector-icons";
import { Button } from "react-native-elements";
import ReadModal from "../components/common/ReadModal";
import Icon from 'react-native-vector-icons/AntDesign';
// import booksData from './books.json';

const BookDetails = (props) => {
  const [book, setBook] = useState({})
  const [user, setUser] = useState({})
  const [liked, setLiked] = useState(false)
  const [loading, setLoading] = useState(true)
  const [modalVisible, setModalVisible] = useState(false);    
  const [read, setRead] = useState("")


  const navigation = useNavigation();
  let id = props.route.params.id
//   let nav = props.route.params.nav
  // let book = booksData.filter(b=>b['ISBN'] == id)[0]
  async function fetchData (){
    await fetch(`${HOST}/api/books`,{
    // await fetch('https://karanjot1995.pythonanywhere.com/books',{
      method: 'GET',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      }
    })
    .then((response) => response.json())
    .then((data) => {
      let b = data.filter(b=>b['id'] == id)[0]
      b.genres = JSON.parse(b.genres)
      setBook(b)
      setLoading(false)

      let user = JSON.parse(localStorage.getItem('user'))
      if(user.shelf[b.id]){
        setRead(user.shelf[b.id])
      }

      if(user.liked && user['liked'].indexOf(String(b['id'])) >-1){
        setLiked(true)
      }
    });
  }

  async function like(){
    let like = !liked
    setLiked(!liked)
    let body = {
      id:book['id'],
      liked: like
    }
    let opts = {
      method: "POST",
      body:JSON.stringify(body),
      headers: {
        Accept: 'application/json',
        'x-access-token': localStorage.getItem('token'),
        Authorization: 'Bearer ' + localStorage.getItem('token'),
        'Content-Type': 'application/json',
      }
    }      
    await fetch(`${HOST}/api/like`, opts)
    // await fetch('http://10.219.175.225:5000/like', opts)
    .then(res=>res.json())
    .then(data=>{
      localStorage.setItem('user', JSON.stringify(data.user))
    })

  }
  const isModalVisible = (visible)=>{
    if(visible){
      setModalVisible(true)
    }else{
      setModalVisible(false)
    }
  }

  const readStatus = async (readStatus) => {
    let body = {};
    body[readStatus.replace(/\s+/g, '_').toLowerCase()] = book.id
    let opts = {
      method: "POST",
      body:JSON.stringify(body),
      headers: {
        Accept: 'application/json',
        'x-access-token': localStorage.getItem('token'),
        Authorization: 'Bearer ' + localStorage.getItem('token'),
        'Content-Type': 'application/json',
      }
    }      
    await fetch(`${HOST}/api/shelf`, opts)
    // await fetch('http://10.219.175.225:5000/like', opts)
    .then(res=>res.json())
    .then(data=>localStorage.setItem('user', JSON.stringify(data.user)))
    setRead(readStatus)
  }

  useEffect(() => {
    fetchData()
    let u = localStorage.getItem('user')
    setUser(u)
  },[])


  let imgUrl = book["cover_img"]
  if(!imgUrl){
    imgUrl = 'https://img.webnovel.com/bookcover/12301378806233905/300/300.jpg?updateTime=1578397501796'
  }
  if(!loading){
  return (
    <ScrollView style={styles.card}>
    {/* <Button title="Go to Page" onPress={ () => navigation.navigate(nav)} /> */}
      <View style={styles.container}>

          <Image 
            style={styles.thunmbNail}
            source={{uri:imgUrl}}
          />
          <View style={{'alignItems':'center'}}>
            <Text style={styles.title}>{book["title"]}</Text>
            <Text style={styles.text}>Author: {book["author"]}</Text>
            <Text style={styles.text}>Publisher: {book["publisher"]}</Text>
            <Text style={styles.text}>Year of Publication: {book["publication_year"]}</Text>
            <Text style={styles.text}>Pages: {book["pages"]}</Text>
            <Text style={styles.text}>Genre: {book["genres"].join(', ')}</Text>
          </View>

          <Pressable onPress={like}>
            <MaterialCommunityIcons
              name={liked ? "heart" : "heart-outline"}
              size={32}
              color={liked ? "red" : "black"}
            />
          </Pressable>
          <Pressable
            style={read?styles.readBtnActive:styles.readBtn}
            onPress={() => setModalVisible(true)}
          >
            {read?<Text style={styles.textStyle}>{read} <Icon name={'edit'} color={'white'} size={24} /></Text>:<Text style={styles.textStyle}>Want to read?</Text>}
            {/* <Text style={styles.textStyle}>{read}</Text> */}
          </Pressable>
          
          <ReadModal read={read} readStatus={readStatus} modalVisible={modalVisible} isModalVisible={isModalVisible} />
          
      </View>
    </ScrollView>
  )}else{
    return (
      <View>
        <Text style={{marginTop:30, textAlign:'center'}}>Loading...</Text>
      </View>
    )
  }
};

const styles = StyleSheet.create({
    container:{
        marginTop:20,
        marginBottom:50,
        padding:10,
        justifyContent: 'center',
        alignItems: 'center',
    },
    thunmbNail:{
      alignSelf: 'center',
      alignItems:'center',
      width: win.width/2.2, 
      height:230
    },
    title:{
        fontSize: 25,
        padding:5,
        paddingTop:20,
        fontWeight:'bold',
        textAlign:'center',
      //   color:'#fff',
    },
    text: {
      fontSize: 20,
      padding:5,
    //   color:'#fff',
      alignItems: 'center',
    },
    button: {
      backgroundColor:"transparent",
      color:'#fff',
//      border:"none",
    //   width:"30%",
      margin:5,
      flexDirection:"row"
    },
    incdec: {
      alignItems: 'center',
    },
    center:{
      textAlign:"center"
    },
    readBtn: {
      backgroundColor: "#2196F3",
      textAlign:'center',
      borderRadius: 10,
      margin:5,
      paddingHorizontal: 10,
      paddingVertical: 5,
      elevation: 2
    },
    readBtnActive: {
      backgroundColor: "#575757",
      textAlign:'center',
      borderRadius: 10,
      margin:5,
      paddingHorizontal: 10,
      paddingVertical: 5,
      elevation: 2
    },
    textStyle: {
      color: "#fff",
      fontSize:20,
    }
})


export default BookDetails