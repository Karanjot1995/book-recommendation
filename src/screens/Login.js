// import { StatusBar } from "expo-status-bar";
import React, { useState } from "react";
import {useNavigation} from '@react-navigation/native';

import {
  StyleSheet,
  Text,
  View,
  Image,
  TextInput,
  Button,
  TouchableOpacity,
  Pressable,
} from "react-native";
import useToken from '../components/UseToken'

 
export default function Login() {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [token, setToken] = useState("")
  const navigation = useNavigation();

  // const { token, removeToken, setToken } = useToken();

  let reg = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w\w+)+$/;

  // if (reg.test(email) === true) {
  //   console.log("Valid");
  // }

  async function login(){
    let body = {
      email:email,
      password: password
    }

    let opts = {
      method: "POST",
      body:JSON.stringify(body),
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      }
    }
    // https://karanjot1995.pythonanywhere.com
    
    await fetch('http://10.219.175.225:5000/token', opts).then(res=>{
      if(res.status==200){
        return res.json()
      }else{
        alert('Incorrect email or password!')
      }
    }).then(data=>{
      localStorage.setItem('token', data.access_token);
      localStorage.setItem('user', JSON.stringify(data.user));
      setToken(data.access_token)
      if(data.access_token){
        navigation.navigate('Tab Navigator')
      }
    })
  }

  return (
    <View style={styles.container}>
      {/* <Image style={styles.image} source={require("./assets/log2.png")} /> */}
 
      {/* <StatusBar style="auto" /> */}
      <View style={styles.inputView}>
        <TextInput
          style={styles.Input}
          placeholder="Email"
          placeholderTextColor="#000"
          autoCapitalize="none" 
          autoCorrect={false}
          onChangeText={(email) => setEmail(email)}
        />
      </View>
 
      <View style={styles.inputView}>
        <TextInput
          style={styles.Input}
          placeholder="Password"
          placeholderTextColor="#000"
          secureTextEntry={true}
          onChangeText={(password) => setPassword(password)}
        />
      </View>
 
      <TouchableOpacity>
        <Text style={styles.forgot_button}>Forgot Password?</Text>
      </TouchableOpacity>
 
      {/* <Pressable style={styles.submit} onPress={login}>
        <Text style={styles.btnText}>LOGIN</Text>
      </Pressable> */}
      <TouchableOpacity onPress={login} style={styles.submit}>
        <Text style={styles.btnText}>LOGIN</Text>
      </TouchableOpacity>

      <TouchableOpacity onPress={()=>navigation.navigate('Sign Up')}>
        <Text style={{marginTop:20, fontSize:20}}>Dont have an account? Sign Up</Text>
      </TouchableOpacity>
    </View>
  );
}
 
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },

 
  inputView: {
    backgroundColor: "#e0e0e0",
    borderRadius: 30,
    width: "70%",
    height: 45,
    marginBottom: 20,
    alignItems: "center",
    // textAlign:'center'
  },
 
  Input: {
    height: 50,
    flex: 1,
    padding: 10,
    textAlign:'center'
    // marginLeft: 20,
  },
 
  forgot_button: {
    height: 30,
    marginBottom: 30,
  },
 
  submit: {
    width: 150,
    borderRadius: 25,
    height: 50,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "#0e0e0e",
    paddingVertical: 12,
    paddingHorizontal: 32,
  },

  btnText:{
    color:'#fff'
  }
});

// import React,{useReducer, useState} from "react";
// import { Text, StyleSheet,View, Image, Button,TextInput,FlatList } from "react-native";

// const Login = (props) => {
//   // const [counter, setCounter,] = useState(0);
//   const [state, setState] = useState({name:''})
  
//   // const [state, dispatch]= useReducer(reducer,{name:''})
//   const data = [
//     {url:"",text:"Forest"},
//     {url:"",text:"Beach"},
//     {url:"",text:"Mountain"},

//   ]
//   const [text,setText] = useState('');
//   const [pass,setPass] = useState('')

//     return (
//        <View style={styles.incdec}>
//         <Text>Movies Page</Text>
        
//             <View style={{padding: 20}}>
//                 <Text style={{fontSize: 30}}>
//                 Enter Name: {state.name}
//                 </Text>
//                 <TextInput
//                 style={styles.input}
//                 placeholder="Type your name here"
//                 // onChangeText={(text)=> dispatch({type:'name', text:text})}
//                 defaultValue={state.name}
//                 />
//                 <Text style={{fontSize: 30}}>
//                 Enter Password: 
//                 </Text>
//                 {/* <TextInput
//                 style={styles.input}
//                 autoCapitalize="none"
//                 autoCorrect={false}
//                 placeholder="Enter password"
//                 // onChangeText={pass => setPass(pass)}
//                 defaultValue={pass}
//                 />
//                 <Text>{pass.length>5?'password strength good':'password must be greater than 5 chars'}</Text>
            
//                 <TouchableOpacity  onPress={()=>navigation.navigate('Components')}>
//                 <Text style={{fontSize:30, textAlign:"center",backgroundColor:"#e0e0e0"}}>Login</Text>
//                 </TouchableOpacity>
//                 */}
//             </View>
//         {/* <TextInput  style={{backgroundColor:color}} ></TextInput> */}
//       </View>
   
//   )
// };


// const styles = StyleSheet.create({
//     text:{
//       textAlign:"center"
//         // backgroundColor:color
//     }
  
// })

// export default Login