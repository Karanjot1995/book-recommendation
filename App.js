// import { createAppContainer } from "react-navigation";
// import { createStackNavigator } from "react-navigation-stack";
// import Navigation from "./src/components/Navigation";
import HomeScreen from "./src/screens/HomeScreen";
import Search from "./src/screens/Search";
import React, { useState, useEffect } from "react";
import { NavigationContainer, useNavigation } from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import 'localstorage-polyfill';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import Login from "./src/screens/Login";
import Recommendations from "./src/screens/Recommendations";
import BookDetails from "./src/screens/BookDetails";
import { Button } from "react-native-elements";
import Profile from "./src/screens/Profile";
import SignUp from "./src/screens/SignUp";
import Liked from "./src/screens/Liked";
import Icon from 'react-native-vector-icons/AntDesign';
import BookShelf from "./src/screens/BookShelf";

// import { Linking } from 'expo';
// const prefix = Linking.createURL('/');
const Stack = createStackNavigator();

const Tab = createBottomTabNavigator();

function HomeStack(){

  return (
  <Stack.Navigator>
    <Stack.Screen
      name="Home"
      component={HomeScreen}
      options={{ headerShown: false }}
    />
    <Stack.Screen
      name="Book Details"
      component={BookDetails}
      options={{
        // headerRight: () => (
        //   <Button
        //     onPress={() => navigation.navigate('Home')}
        //     color="#fff"
        //   />
        // ),
      }}
      // options={{ headerShown: false }}
    />
  </Stack.Navigator>
  )

}

function MyTabs() {
  const [token, setToken] = useState("")
  const navigation = useNavigation();

  useEffect(() => {
    let t = localStorage.getItem("token")
    setToken(t)
    if(!t){
      navigation.navigate('Login')
    }
  },[])
  const screenOptions = (route, color) => {
    let iconName;
  
    switch (route.name) {
      case 'Home':
        iconName = 'home';
        break;
      case 'Search':
        iconName = 'search1';
        break;
      case 'Recommended':
        iconName = 'smileo';
        break;
      case 'Liked':
        iconName = 'staro';
        break;
      case 'Book Shelf':
        iconName = 'book';
        break;
      case 'Profile':
        iconName = 'user';
        break;
      default:
        break;
    }
  
    return <Icon name={iconName} color={color} size={24} />;
  };
  return (
    <Tab.Navigator screenOptions={({route}) => ({
      tabBarIcon: ({color}) => screenOptions(route, color),
    })}>
      <Tab.Screen name="Home" component={HomeStack} />
      <Tab.Screen name="Recommended" component={Recommendations} />
      <Tab.Screen name="Liked" component={Liked} />
      <Tab.Screen name="Book Shelf" component={BookShelf} />
      <Tab.Screen name="Profile" component={Profile} />
      <Tab.Screen name="Search" component={Search} />
    </Tab.Navigator>
  );
}

export default function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen
          name="Login"
          component={Login}
          options={{ headerShown: false }}
        /> 
        <Stack.Screen
          name="Sign Up"
          component={SignUp}
          options={{ headerShown: false }}
        /> 
        <Stack.Screen
          name="Tab Navigator"
          component={MyTabs}
          options={{ headerShown: false }}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
};



// function AuthNavigator() {

//   return (
//     <Tab.Navigator>
//       <Tab.Screen name="Home" component={HomeScreen} />
//       <Tab.Screen name="Login" component={Login} />
//       <Tab.Screen name="Search" component={Search} />
//     </Tab.Navigator>
//   );
// }

// export default function App() {
//   const [token, setToken] = useState("")

//   useEffect(() => {
//     setToken(localStorage.getItem("token"))
//   },[])
//   return (
//     <NavigationContainer>
//       {token?<MyTabs />:<AuthNavigator/>}
//     </NavigationContainer>
//   );
// }

// const navigator = createStackNavigator(
//   {
//     Home: {
//       screen: HomeScreen,
//       // navigationOptions: {
//       //   headerTitle: ()=> <Navigation/>
//       // }
//     },
//     Navigation: Navigation,
//     Search: Search,
//     Movies: MoviesScreen
//   },
//   {
//     initialRouteName: "Home",
//     defaultNavigationOptions: {
//       title: "App",
//     },
//     headerMode: 'screen',
//     cardStyle: { backgroundColor: '#141414' },

//   }

// );

// export default createAppContainer(MyTabs);



// AIzaSyDW4UKBQ0_CLrOI2WJTXVrmii9rurxXqrI