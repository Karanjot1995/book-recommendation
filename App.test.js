import React from 'react';
import { Card } from 'react-native-elements';
import renderer from 'react-test-renderer';
import App from './App';
import BookDetails from './src/screens/BookDetails';
import Login from './src/screens/Login';
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

describe('<App />', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<App />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('App test against snapshot', () => {
        const tree = renderer.create(<App />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});


describe('Card', () => {
it('should render correctly', () => {
    const header = renderer.create(
    <Card item={{'cover_img':'https://img.webnovel.com/bookcover/12301378806233905/300/300.jpg?updateTime=1578397501796'}} />
    ).toJSON;
    expect(header).toMatchSnapshot();
});
});
