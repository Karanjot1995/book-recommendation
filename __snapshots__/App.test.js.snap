// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> App test against snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          gestureResponseDistance={
            Object {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                handlerTag={2}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "flex": 1,
                            "justifyContent": "center",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#e0e0e0",
                              "borderRadius": 30,
                              "height": 45,
                              "marginBottom": 20,
                              "width": "70%",
                            }
                          }
                        >
                          <TextInput
                            autoCapitalize="none"
                            autoCorrect={false}
                            onChangeText={[Function]}
                            placeholder="Email"
                            placeholderTextColor="#000"
                            style={
                              Object {
                                "flex": 1,
                                "height": 50,
                                "padding": 10,
                                "textAlign": "center",
                              }
                            }
                          />
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#e0e0e0",
                              "borderRadius": 30,
                              "height": 45,
                              "marginBottom": 20,
                              "width": "70%",
                            }
                          }
                        >
                          <TextInput
                            onChangeText={[Function]}
                            placeholder="Password"
                            placeholderTextColor="#000"
                            secureTextEntry={true}
                            style={
                              Object {
                                "flex": 1,
                                "height": 50,
                                "padding": 10,
                                "textAlign": "center",
                              }
                            }
                          />
                        </View>
                        <View
                          accessible={true}
                          collapsable={false}
                          focusable={false}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "height": 30,
                                "marginBottom": 30,
                              }
                            }
                          >
                            Forgot Password?
                          </Text>
                        </View>
                        <View
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#0e0e0e",
                              "borderRadius": 25,
                              "height": 50,
                              "justifyContent": "center",
                              "opacity": 1,
                              "paddingHorizontal": 32,
                              "paddingVertical": 12,
                              "width": 150,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#fff",
                              }
                            }
                          >
                            LOGIN
                          </Text>
                        </View>
                        <View
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "fontSize": 20,
                                "marginTop": 20,
                              }
                            }
                          >
                            Dont have an account? Sign Up
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;

exports[`<App /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          gestureResponseDistance={
            Object {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                handlerTag={1}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "flex": 1,
                            "justifyContent": "center",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#e0e0e0",
                              "borderRadius": 30,
                              "height": 45,
                              "marginBottom": 20,
                              "width": "70%",
                            }
                          }
                        >
                          <TextInput
                            autoCapitalize="none"
                            autoCorrect={false}
                            onChangeText={[Function]}
                            placeholder="Email"
                            placeholderTextColor="#000"
                            style={
                              Object {
                                "flex": 1,
                                "height": 50,
                                "padding": 10,
                                "textAlign": "center",
                              }
                            }
                          />
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#e0e0e0",
                              "borderRadius": 30,
                              "height": 45,
                              "marginBottom": 20,
                              "width": "70%",
                            }
                          }
                        >
                          <TextInput
                            onChangeText={[Function]}
                            placeholder="Password"
                            placeholderTextColor="#000"
                            secureTextEntry={true}
                            style={
                              Object {
                                "flex": 1,
                                "height": 50,
                                "padding": 10,
                                "textAlign": "center",
                              }
                            }
                          />
                        </View>
                        <View
                          accessible={true}
                          collapsable={false}
                          focusable={false}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "height": 30,
                                "marginBottom": 30,
                              }
                            }
                          >
                            Forgot Password?
                          </Text>
                        </View>
                        <View
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#0e0e0e",
                              "borderRadius": 25,
                              "height": 50,
                              "justifyContent": "center",
                              "opacity": 1,
                              "paddingHorizontal": 32,
                              "paddingVertical": 12,
                              "width": 150,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#fff",
                              }
                            }
                          >
                            LOGIN
                          </Text>
                        </View>
                        <View
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "fontSize": 20,
                                "marginTop": 20,
                              }
                            }
                          >
                            Dont have an account? Sign Up
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;

exports[`Card should render correctly 1`] = `[Function]`;
